project_task_code/models/project_task.py

```diff
@@ -28,3 +29,22 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        res = super(ProjectTask, self).name_search(
+            name=name,
+            args=args,
+            operator=operator,
+            limit=limit)
+        if not args:
+            args = []
+        tasks = self.env['project.task']
+        # Remove here a leading character, if name is not purely digits
+        name = name if name.isdigit() else name.lstrip('T')
+        # We are searching for codes
+        # operator is always 'ilike'
```

Comment:
you don't control that, the caller of the function controls that. And the way I envision this override, it should for the code part always search `('code', '=ilike', name + '%')`

Reply:
(and btw. also sort by code descending, then we'll get higher numbers = newer tasks higher up, which probably is what we want)